otx/cli/tools/build.py

```diff
@@ -57,56 +58,102 @@ def parse_args():
     return parser.parse_args()
 
 
-def main():
-    """Main function for model or backbone or task building."""
-
-    args = parse_args()
-    args.task = args.task.upper() if args.task is not None else args.task
-
-    builder = Builder()
-
-    otx_root = get_otx_root_path()
+# pylint: disable=too-many-arguments
+def build(
+    builder: Builder,
+    train_data_roots: str = None,
+    val_data_roots: str = None,
+    task: str = None,
+    train_type: str = "incremental",
+    workspace_root: str = None,
+    model: str = None,
+    backbone: str = None,
+    save_backbone_to: str = None,
+    otx_root: str = ".",
+    template: ModelTemplate = None,
+):
+    """Makes configuration files (data.yaml, template.yaml, ...) in workspace directory.
+
+    Args:
+        builder (Builder): Builder class
+        train_data_roots (str, optional): The path of training dataset. Defaults to None.
+        val_data_roots (str, optional): The path of validation dataset. Defaults to None.
+        task (str, optional): The name of task (i.e. classification, detection, segmentation, ..). Defaults to None.
+        train_type (str, optional): The type of training (i.e. incremental, semi-sl, ..). Defaults to "incremental".
```

Comment:
Are both `semi-sl` and `semisl` supported?

Reply:
I'm sorry, I can't understand what you want to said

Reply:
In my doc-string, I just typed `semi-sl`

Reply:
I think it's a typo error. Isn't `semisl` correct?

Reply:
I was talking about this.
![image](https://user-images.githubusercontent.com/38045080/215715611-68fb0ee2-45d9-4429-ac78-7eb94a1e6f96.png)
We have to pick one of these.

Reply:
Sorry for confusing comment, absolutely same opinion with Harim.

Reply:
Understood, will edit this